src/bdf_utils.jl

```diff
@@ -69,4 +70,26 @@ function reinterpolate_history!(cache::OrdinaryDiffEqConstantCache, D, R, k)
   end
 end
 
-global const γₖ = @SVector[sum(1//j for j in 1:k) for k in 1:6]
\ No newline at end of file
+
+function calc_R(ρ, k, ::Val{N}) where {N}
+  R = zero(MMatrix{N,N,typeof(ρ)})
+  @inbounds for r = 1:k
+    R[1,r] = -r * ρ
+    for j = 2:k
+      R[j,r] = R[j-1,r] * ((j-1) - r * ρ)/j
+    end
+  end
+  SArray(R)
+end
+
+function update_D!(D, dd, k)
+  dd = _vec(dd)
+  @views @.. D[:,k+2] = dd - D[:,k+1]
+  @views @.. D[:,k+1] = dd
+  for i in k:-1:1
+    @views @.. D[:,i] = D[:,i] + D[:,i+1]
+  end
+  return nothing
+end
+
+const γₖ = @SVector[sum(1//j for j in 1:k) for k in 1:6]
```

Comment:
Actually, due to https://github.com/JunpengGao233/OrdinaryDiffEq.jl/blob/cc45e743e9505809d268c9c2d4ecd5344eba80be/src/caches/bdf_caches.jl#L290 and a similar line for the mutable cache, I am wondering if this line is needed at all.

Reply:
This is needed for the old QNDF1 and QNDF2 methods